### PR TITLE
Bob/4656 refactor test order service

### PR DESCRIPTION
**Pull request by fzhao99**
# BACKEND PULL REQUEST

## Related Issue

- Fixes the core pieces of #4645 

## Changes Proposed

- Removes the `addTestResult` methods and replaces them with multiplex methods in the test class
- Updates the relevant test assertions needed

## Additional Information
- Added a helper method to generate the multiplex result because I got tired of copy pasting the same lines of code over and over again but lmk if the method / overload is unclear
- Sonar is complaining about us using some deprecated methods in the [TestEventRepository](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java). I dug into it briefly, but for scope reasons won't address them here. Happy to follow up with another ticket to clean those up alongside splitting up the TestOrderService into smaller pieces as discussed in #4645 

## Testing

- Verify test changes are no-ops (special attention to the size assertions changes in `addTestResult_correction` and `addTestResult_savesResults`)

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java [hunk 1]
```diff
@@ -594,9 +614,9 @@ void addTestResult_correction() {
         LocalDate.of(1865, 12, 25),
         LocalDate.ofInstant(
             correctionTestEvent.getDateTested().toInstant(), ZoneId.systemDefault()));
-    assertEquals(2, _resultRepository.findAllByTestOrder(response.getTestOrder()).size());
-    assertEquals(1, _resultRepository.findAllByTestEvent(correctionTestEvent).size());
-    assertEquals(1, _resultRepository.findAllByTestEvent(originalTestEvent).size());
+    assertEquals(6, _resultRepository.findAllByTestOrder(response.getTestOrder()).size());
```
Comment (fzhao99) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java:617]: Not sure if we actually want the test result size assertions to get multiplied by 3 here. Lmk If I should be constructing the test events in another way!

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java [hunk 2]
```diff
@@ -825,13 +851,14 @@ void addTestResult_savesResults() {
...
     AddTestResultResponse res =
-        _service.addTestResult(
-            devA.getInternalId(), TestResult.POSITIVE, patient.getInternalId(), null);
+        _service.addMultiplexResult(
+            devA.getInternalId(), testResultInput, patient.getInternalId(), null);
 
     // THEN
     List<Result> results = _resultRepository.findAllByTestOrder(res.getTestOrder());
-    assertEquals(1, results.size());
+    assertEquals(3, results.size());
```
Comment (fzhao99) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java:861]: Ditto to the [above comment.](https://github.com/CDCgov/prime-simplereport/pull/4760/files#r1029726095)

**Review by georgehager — COMMENTED** [review]
overall lgtm! just a couple comments for readability.

**Review by emmastephenson — COMMENTED** [review]
This is a good start, but I think you need to reconsider the tests slightly - in most cases you're changing the underlying assertions from a covid-only test to a multiplex test, but I would have a preference to continue testing the covid-only case. You should still be able to use the `addMultiplexResult`, you'll just need to change the input from a multiplex input to a covid-only one. Let me know if this doesn't make sense; happy to chat offline!

**Review by emmastephenson — APPROVED** [review]
LGTM, thanks for changing the tests!

**Pull request merged**